mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical shadow estimation for quantum circuits. Based on the paper"""
+
+from typing import (
+    Optional,
+    Callable,
+    List,
+    Dict,
+    Any,
+    Tuple,
+    Union,
+    Mapping,
+)
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+
+import mitiq
+from mitiq import MeasurementResult
+from mitiq.shadows.quantum_processing import random_pauli_measurement
+from mitiq.shadows.classical_postprocessing import (
+    shadow_state_reconstruction,
+    expectation_estimation_shadow,
+    get_pauli_fidelities,
+)
+
+
+def pauli_twirling_calibrate(
+    k_calibration: int = 1,
+    locality: Optional[int] = None,
+    zero_state_shadow_outcomes: Optional[Tuple[List[str], List[str]]] = None,
+    qubits: Optional[List[cirq.Qid]] = None,
+    executor: Optional[Callable[[cirq.Circuit], MeasurementResult]] = None,
+    num_total_measurements_calibration: Optional[int] = 20000,
+) -> Dict[str, complex]:
+    r"""
+    This function returns the dictionary of the median of means estimation
+    of Pauli fidelities: {:math:`\{'b':f_{b}\}_{b\in\{0,1\}^n}`}.
+
+    Args:
+        k_calibration: Number of groups of "median of means" used for
+            calibration.
+        locality: The locality of the operator, whose expectation value is
+            going to be estimated by the classical shadow. e.g. if operator is
+            Ising model Hamiltonian with nearist neighbour interacting, then
+            locality = 2.
+        zero_state_shadow_outcomes: The output of function
+            `shadow_quantum_processing` of zero calibrate state.
```

Comment:
If you want to reference a function, and have a link show up in the API-doc you can use the syntax `` :func:`function_name` `` to do so. See an example [here](https://github.com/unitaryfund/mitiq/blob/891684b29519afc37801c1a563913345e13365a8/mitiq/pec/representations/depolarizing.py#L180).

Could be useful here to link to `shadow_quantum_processing`.

Reply:
thanks! will add it!